src/main/java/com/uber/profiling/Profiler.java

```diff
@@ -16,10 +16,45 @@
 
 package com.uber.profiling;
 
-public interface Profiler {
-    long getIntervalMillis();
+import com.uber.profiling.util.AgentLogger;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.atomic.AtomicLong;
 
-    void setReporter(Reporter reporter);
+public abstract class Profiler implements Runnable {
```

Comment:
Profiler now is a `Runnable` so it can be:
- either run one-off if it's not scheduled run periodically
- or, scheduled to run periodically